python/taichi/lang/impl.py

```diff
@@ -141,11 +142,15 @@ def create_program(self):
         if self.prog is None:
             self.prog = taichi_lang_core.Program()
 
+    def try_materialize(self):
+        if not self.layout_materialized:
+            self.materialize_layout_callback()
```

Comment:
@yuanming-hu What's the difference between directly `self.materialize` which has a `if not` guard too, isn't it? What's the point of `Expr.layout_materialized` at the first place?

Reply:
Why not response me so long :(

Reply:
Sorry about the delayed reply. @archibate could you clarify your question? I'm not sure if I get you correctly.

Reply:
Why you use a separate variable `Expr.layout_materialized` instead of simple use the existing `ti.get_runtime().materialized`? What the point of this design at first place?

Reply:
I see - good point. I don't remember why I did that TBH. Maybe we should try to remove `Expr.layout_materilized` and use `ti.get_runtime().materialized` instead, and see if anything goes wrong (of course, in a future PR).

Reply:
OK, I will. So far `ti.get_runtime().layout_materialized` is good enough in this PR, right? Or you want to revert that now?